Fix BlacklistPasswordPolicyDefaultPath Failures on Windows

### Describe the bug

password-blacklists are failing for Windows environment 

Could be related to https://issues.redhat.com/browse/KEYCLOAK-19463

### Version

18.0.0

### Expected behavior

Tests should pass successfully.

### Actual behavior

Tests are failing with the following error

`java.lang.AssertionError: 

Expected: a string ending with "/data/password-blacklists/"
     but: was "C:\Jenkins\workspace\universal-tes---ea900cfa\jvm_tests_win-win2016-win_openjdk8\keycloak\testsuite\integration-arquillian\tests\base\target\containers\auth-server-wildfly\standalone\data/password-blacklists/"`

### How to Reproduce?

_No response_

### Anything else?

_No response_

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/policy/QuarkusBlacklistPasswordPolicyProviderFactory.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/policy/QuarkusBlacklistPasswordPolicyProviderFactory.java
@@ -20,6 +20,8 @@ package org.keycloak.quarkus.runtime.policy;
 import org.keycloak.policy.BlacklistPasswordPolicyProviderFactory;
 import org.keycloak.quarkus.runtime.Environment;
 
+import java.io.File;
+
 /**
  * <p>Quarkus implementation of the BlacklistPasswordPolicyProviderFactory. The
  * default path for the list files is calculated using the quarkus environment
@@ -31,6 +33,6 @@ public class QuarkusBlacklistPasswordPolicyProviderFactory extends BlacklistPass
 
     @Override
     public String getDefaultBlacklistsBasePath() {
-        return Environment.getDataDir() + "/" + PASSWORD_BLACKLISTS_FOLDER;
+        return Environment.getDataDir() + File.separator + PASSWORD_BLACKLISTS_FOLDER;
     }
 }

--- a/server-spi-private/src/main/java/org/keycloak/policy/BlacklistPasswordPolicyProviderFactory.java
+++ b/server-spi-private/src/main/java/org/keycloak/policy/BlacklistPasswordPolicyProviderFactory.java
@@ -25,6 +25,7 @@ import org.keycloak.models.KeycloakSession;
 import org.keycloak.models.KeycloakSessionFactory;
 
 import java.io.BufferedReader;
+import java.io.File;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
@@ -75,7 +76,7 @@ public class BlacklistPasswordPolicyProviderFactory implements PasswordPolicyPro
 
     public static final String JBOSS_SERVER_DATA_DIR = "jboss.server.data.dir";
 
-    public static final String PASSWORD_BLACKLISTS_FOLDER = "password-blacklists/";
+    public static final String PASSWORD_BLACKLISTS_FOLDER = "password-blacklists" + File.separator;
 
     private ConcurrentMap<String, FileBasedPasswordBlacklist> blacklistRegistry = new ConcurrentHashMap<>();
 
@@ -142,7 +143,7 @@ public class BlacklistPasswordPolicyProviderFactory implements PasswordPolicyPro
      * when no other configuration is in place.
      */
     public String getDefaultBlacklistsBasePath() {
-        return System.getProperty(JBOSS_SERVER_DATA_DIR) + "/" + PASSWORD_BLACKLISTS_FOLDER;
+        return System.getProperty(JBOSS_SERVER_DATA_DIR) + File.separator + PASSWORD_BLACKLISTS_FOLDER;
     }
 
     /**

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/policy/PasswordPolicyTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/policy/PasswordPolicyTest.java
@@ -35,6 +35,7 @@ import org.keycloak.testsuite.arquillian.annotation.AuthServerContainerExclude.A
 import org.keycloak.testsuite.util.ContainerAssume;
 import org.keycloak.testsuite.util.RealmBuilder;
 
+import java.io.File;
 import java.util.List;
 
 import static org.hamcrest.CoreMatchers.instanceOf;
@@ -192,12 +193,14 @@ public class PasswordPolicyTest extends AbstractKeycloakTest {
 
     @Test
     public void testBlacklistPasswordPolicyDefaultPath() throws Exception {
+        final String SEPARATOR = File.separator;
+
         testingClient.server("passwordPolicy").run(session -> {
             ProviderFactory<PasswordPolicyProvider> passPolicyFact = session.getKeycloakSessionFactory().getProviderFactory(
                     PasswordPolicyProvider.class, BlacklistPasswordPolicyProviderFactory.ID);
             assertThat(passPolicyFact, instanceOf(BlacklistPasswordPolicyProviderFactory.class));
-            assertThat(((BlacklistPasswordPolicyProviderFactory)passPolicyFact).getDefaultBlacklistsBasePath(),
-                    endsWith("/data/password-blacklists/"));
+            assertThat(((BlacklistPasswordPolicyProviderFactory) passPolicyFact).getDefaultBlacklistsBasePath(),
+                    endsWith(SEPARATOR + "data" + SEPARATOR + "password-blacklists" + SEPARATOR));
         });
     }
 

